graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java

```diff
@@ -206,9 +210,12 @@ public IndexSetStats globalStats() {
     public IndexSetSummary get(@ApiParam(name = "id", required = true)
                                @PathParam("id") String id) {
         checkPermission(RestPermissions.INDEXSETS_READ, id);
+        final IndexSet indexSet = indexSetRegistry.get(id).orElseThrow(() -> new NotFoundException("Couldn't find index set with ID <" + id + ">"));
         final IndexSetConfig defaultIndexSet = indexSetService.getDefault();
         return indexSetService.get(id)
-                .map(config -> IndexSetSummary.fromIndexSetConfig(config, config.equals(defaultIndexSet)))
+                .map(config -> IndexSetSummary.fromIndexSetConfig(
+                        config, config.equals(defaultIndexSet),
+                        snapshotService.getFailedSnapshotName(indexSet, config)))
```

Comment:
It seems to me this is the reason for all of the rest of the code to be in graylog2-server.

A better solution would be to create a more generic extension point for contributing index set information from plugins, instead of pulling the very specific snapshot implementation into the server code base.
As Othello noted, a more generic data tiering info endpoint that's empty by default is probably the better choice.

Reply:
> a more generic data tiering info endpoint that's empty by default 


@kay Makes sense. Can we make this refactoring an issue for cooldown? 
We're running out of time to merge this in the current cycle and I hate rushing it.

Reply:
I'd greatly prefer not to merge code we don't intend to keep, to be honest.

Reply:
Cool - to expedite this, it'd be very helpful if you could spec this generic endpoint.

Reply:
The mechanics are the same as the warm-tier configuration, basically a dummy implementation like SnapshotServiceNoop.
There's multiple options here:
* Either include a more generic property in IndexSetSummary, e.g. `data_tiering` and return a specific hot/warm object by calling data tiering summary implementation. The the hot/warm specific UI can use that to render the button if necessary and call the enterprise endpoint for deleting the snapshot, or whatever it needs to do. This way it's separated for any future data tiering implementation without any additional work required in the server
* Or entirely do it from the UI, loading (possibly in bulk) hot/warm details for each index set, that then returns the specific object, e.g. in this case the list of failed/partial snapshot information to let the UI render it.